static/app/components/alerts/snoozeAlert.tsx

```diff
@@ -23,12 +24,12 @@ type Props = {
   ruleId: string;
 };
 
-function SnoozeAlert({isSnoozed, onSnooze, projectSlug, ruleId}: Props) {
+function SnoozeAlert({isSnoozed, onSnooze, projectSlug, ruleId, hasAccess}: Props) {
   const organization = useOrganization();
   const api = useApi();
   const location = useLocation();
 
-  const [disabled, setDisabled] = useState(false);
+  const [disabled, setDisabled] = useState(!hasAccess);
```

Comment:
i don't think access makes sense to set as the default here since it won't be reactive as props are expected to be.

instead you might just set the button to disabled w/ `disabled={disabled || !hasAccess}`